### ISSUE TITLE
Zr.require is not a function (React - Bun Dev Error)

### ISSUE THREAD

**Issue by jesseagleboy** [oven-sh/bun#3105]
### What version of Bun is running?

0.6.4

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

1. Create an HTML file (either in the main folder or inside of a public/ directory)
2. Create a `.jsx` file that initializes a React component to the DOM (querying the id on the HTML and using createRoot).
3. Add a script tag to the HTML file with the src equal to the `.jsx` file.
4. Run `bun dev`.
5. Browser console error should come back saying `Zr.require is not a function`.

### What is the expected behavior?

React should properly load and be applied to the HTML document.

### What do you see instead?

I had test projects set up from before `0.6.3` where running `bun dev` worked with React, but now there seems to be some type-error going on. The HTML page itself loads but the div being target for React will not appear and a console log error will come up.

### Additional information

The stack trace shows two files:
1. `index.js:6:28` of react/index - this points to `module.exports = require('./cjs/react.development.js');`
2. `bun:wrap:7:7847` - this points to `Nr = e=>e != null && typeof e == "object" && (e.default && e.default[de] || e[de]), Fr = e=>Zr.require(e), jr = ur, Vr = (e,r)=>(Object.defineProperty(e, "name", {...`

I'm assuming that with very minimal code React could be up and running on `bun dev` with the steps I've taken above. But I know to look out for breaking changes so I can understand if there are changes of what imports are happening with Bun now.

**Comment by steambap** [comment]
Same issue still exists in 0.6.7

**Comment by mi4uu** [comment]
v0.6.14  same thing, anything we can do abut it?

**Comment by jesseagleboy** [comment]
> v0.6.14 same thing, anything we can do abut it?

I think there was another submitted issue about something similar a little bit ago. Someone replied that the `dev` command was being redone. Therefore, there hasn't been any changes in these releases. It seems the only thing so far to do is use a template like the react-bun or run your own server that does a build.

**Comment by zapling** [comment]
This issue still occurs in `0.8.0`

**Comment by Vishtar** [comment]
`0.8.1`, still.

**Comment by jesseagleboy** [comment]
@zapling @Vishtar I don't think they have released a fix for the native `dev` command for Bun for some time. It seems that the current option for an accessible server with Bun is using `Vite`. Unless things like React can work with `bun dev`, Vite can still do React with little dev effort. I've managed to use Svelte and React on the same page using Vite and also using the `bunx --bun vite` command (I also was able to easily pass data between them using `Preact Signals` which is something it seems using the `bun build` command is incompatible with but the regular `vite build` handles fine). Link: https://bun.sh/guides/ecosystem/vite
